src/tip-opengov.e2e.ts

```diff
@@ -25,6 +26,22 @@ const treasuryAccount = "13UVJyLnbVp9RBZYFwFGyDvVd1y27Tt8tkntv6Q7JVPhFsTB"; // h
 
 const network = "localrococo";
 
+const signAndSend = (signer: KeyringPair, extrinsic: SubmittableExtrinsic<"promise">) =>
+  new Promise<void>(async (resolve, reject) => {
+    try {
+      await extrinsic.signAndSend(signer, { nonce: -1 }, (result) => {
+        if (result.isError) {
+          reject(result.status.toString());
+        }
+        if (result.isInBlock) {
+          resolve();
+        }
+      });
+    } catch (e) {
+      reject(e);
+    }
+  });
```

Comment:
Why do we use two types of following the status (awaiting on result, and using the status callback) at the same time here?

Reply:
One status (awaiting on result) is the status of sending the extrinsic. It can fail e.g. if we can't connect to the RPC.
The second status (status callback) is the status of the extrinsic on the chain.

Reply:
I was under impression that `await signAndSend` would wait for extrinsic to end up in the best block...  
When does it actually return?

Reply:
> I was under impression that `await signAndSend` would wait for extrinsic to end up in the best block...

Unfortunately not.

> The result for this call (we will deal with subscriptions in a short while), is the transaction hash. This is a hash of the data and receiving this does not mean that transaction has been included, but rather only that it has been accepted for propagation by the node. (It can still fail on execution, we will handle this in some of our follow-up sections.)

From the [docs](https://polkadot.js.org/docs/api/start/api.tx).

Reply:
huh, okay